doc/developer-notes.md

```diff
@@ -5,8 +5,10 @@ Developer Notes
 **Table of Contents**
 
 - [Developer Notes](#developer-notes)
-    - [Coding Style](#coding-style)
+    - [Coding Style (General)](#coding-style-general)
+    - [Coding Style (C++)](#coding-style-c)
     - [Doxygen comments](#doxygen-comments)
```

Comment:
nit: If I’m not mistaken we’re only using doxygen for C++, how about making this a sub-point of the previous, here and below?